File: test_vuln.py
  import subprocess                                                                            
  import sqlite3
  import os                                                                                    
                                                            
  def sql_injection_vuln(username, password):                                                  
      # SQL injection vulnerability                                                            
      query = f"SELECT * FROM users WHERE username='{username}' AND password='{password}'"     
      conn = sqlite3.connect('database.db')                 
      cursor = conn.cursor()                                                                   
      cursor.execute(query)
      return cursor.fetchone()                                                                 
                                                            
  def command_injection_vuln(user_file):                                                       
      # Command injection vulnerability                     
      os.system(f"cat /var/log/{user_file}")                                                 
      subprocess.call(f"grep error {user_file}", shell=True)
                                                                                               
  def path_traversal_vuln(filename):
      # Path traversal vulnerability                                                           
      with open(f"/uploads/{filename}", 'r') as f:          
          return f.read()                                                                      
  EOF 

File: test_vuln.js
const mysql = require('mysql');
const connection = mysql.createConnection({ ... });

function login(username, password) {
    // SQL Injection
    const query = `SELECT * FROM users WHERE username = '${username}' AND password = '${password}'`;
    connection.query(query, (err, results) => {
        // ...
    });
}

function runCommand(cmd) {
    // Command Injection
    const { exec } = require('child_process');
    exec(cmd, (err, stdout, stderr) => {
        // ...
    });
}
